core/src/main/scala/org/http4s/Status.scala

```diff
@@ -77,51 +77,46 @@ object Status {
     val ServerError = Status.Informational
   }
 
-  val MinCode = 100
-  val MaxCode = 599
+  private[http4s] val MinCode = 100
+  private[http4s] val MaxCode = 599
 
-  def fromInt(code: Int): ParseResult[Status] = lookup(code) match {
-    case None => parseAsStatus(code)
-    case Some(status) => Right(status)
+  def fromInt(code: Int): ParseResult[Status] = withRangeCheck(code) {
+    val lookupResult = lookup(code)
+    if (lookupResult.isRight) lookupResult
+    else ParseResult.success(Status(code, "No reason provided"))
   }
 
-  def fromIntAndReason(code: Int, reason: String): ParseResult[Status] =
-    lookup(code, reason) match {
-      case None => parseAsStatus(code, reason)
-      case Some(status) => Right(status)
-    }
+  def fromIntAndReason(code: Int, reason: String): ParseResult[Status] = withRangeCheck(code) {
+    val lookupResult = lookup(code, reason)
+    if (lookupResult.isRight) lookupResult else ParseResult.success(Status(code, reason))
```

Comment:
The match can still be done without allocating, if you like.  (It's all about the same to me.)

```scala
lookup(code, reason) match {
  case right: Right => right
  case _: Left => ParseResult.success(Status(code, reason))
}
```